nixos/matrix-synapse: configure PYTHONMALLOC when using jemalloc

mweinelt opened the pull request:
Per a synapse FAQ entry[1] this should be set in combination with\ jemalloc to make sure that jemalloc can accurately calculate memory usage.

[1] https://github.com/element-hq/synapse/blob/develop/docs/usage/administration/admin_faq.md#help-synapse-is-slow-and-eats-all-my-ramcpu


<!--
^ Please summarise the changes you have done and explain why they are necessary here ^

For package updates please link to a changelog or describe changes, this helps your fellow maintainers discover breaking updates.
For new packages please briefly describe the package or provide a link to its homepage.
-->

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

mweinelt committed:
nixos/matrix-synapse: configure PYTHONMALLOC when using jemalloc

Per a synapse FAQ entry[1] this should be set in combination with\
jemalloc to make sure that jemalloc can accurately calculate memory
usage.

[1] https://github.com/element-hq/synapse/blob/develop/docs/usage/administration/admin_faq.md#help-synapse-is-slow-and-eats-all-my-ramcpu

mweinelt commented:
I also still think we should enable jemalloc usage by default. The official docker image has been doing it for years.

https://github.com/element-hq/synapse/blob/ebc21a8c67ed7a05af010e329b6402cd41229ebe/docker/start.py#L235-L240

Ma27 commented:
> I also still think we should enable jemalloc usage by default. The official docker image has been doing it for years.

I'm using for quite a while as well. If it's the upstream default anyways, I'd say go for it :)

mweinelt committed:
nixos/matrix-synapse: enable jemalloc by default

The official docker image has been defaulting to the jemalloc memory
allocator for years.

999eagle reviewed (approved):
lgtm. matrix has been running with jemalloc for years here as well; defaulting to that should be fine.
